src/data/bedrock/block/convert/BlockStateSerializerHelper.php

```diff
@@ -288,4 +290,15 @@ public static function encodeWoodenSlab(Slab $block, string $typeValue) : BlockS
 		return self::encodeSlab($block, Ids::WOODEN_SLAB, Ids::DOUBLE_WOODEN_SLAB)
 			->writeString(BlockStateNames::WOOD_TYPE, $typeValue);
 	}
+
+	public static function encodeChiseledBookshelf(ChiseledBookshelf $block) : BlockStateWriter{
+		$slots = array_keys($block->getItems());
```

Comment:
This `array_keys()` calls is redundant. A `foreach($block->getItems() as $slot => $item)` can be used instead.